src/http/filters/jwt_filter.rs

```diff
@@ -0,0 +1,109 @@
+use crate::models::token::BoxerToken;
+use crate::models::validation_settings::ValidationSettings;
+use actix_web::dev::{forward_ready, Service, ServiceRequest, ServiceResponse, Transform};
+use actix_web::error::ErrorUnauthorized;
+use actix_web::Error;
+use futures_util::future::LocalBoxFuture;
+use jwt_authorizer::{Authorizer, JwtAuthorizer, Validation};
+use log::{debug, error};
+use std::sync::Arc;
+
+/// Middleware for external token validation factory
+pub struct InternalTokenMiddlewareFactory {}
+
+/// The ExternalTokenMiddlewareFactory's own methods implementation
+impl InternalTokenMiddlewareFactory {
+    pub(crate) fn new() -> Self {
+        InternalTokenMiddlewareFactory {}
+    }
+}
+
+/// Transform trait implementation
+/// `NextServiceType` - type of the next service
+/// `BodyType` - type of response's body
+impl<NextService, BodyType> Transform<NextService, ServiceRequest>
+    for InternalTokenMiddlewareFactory
+where
+    NextService:
+        Service<ServiceRequest, Response = ServiceResponse<BodyType>, Error = Error> + 'static,
+    NextService::Future: 'static,
+    BodyType: 'static,
+{
+    type Response = ServiceResponse<BodyType>;
+    type Error = Error;
+    type Transform = JwtAuthorizerMiddleware<NextService>;
+    type InitError = ();
+    type Future =
+        LocalBoxFuture<'static, Result<JwtAuthorizerMiddleware<NextService>, Self::InitError>>;
+
+    fn new_transform(&self, service: NextService) -> Self::Future {
+        Box::pin(async move {
+            let settings = ValidationSettings::new();
+            let mut validation = Validation::new();
+            validation.iss = Some(settings.valid_issuers.clone());
+            validation.aud = Some(settings.valid_audiences.clone());
+
+            // It's OK to unwrap here because we should panic if cannot build the authorizer
+            let authorizer = JwtAuthorizer::from_secret(settings.secret)
+                .validation(validation)
+                .build()
+                .await
+                .unwrap();
```

Comment:
```suggestion
                 .expect("Failed to build JwtAuthorizer.");
```
I would suggest handling the errors explicitly like handling it with a error message or something similar. Not that important, just a suggestion 🙂

Reply:
Thanks, Andre! It's a very good suggestion 👍